### ISSUE TITLE
Document how to write autograd-supported PyTorch function with CuPy

### ISSUE THREAD

**Issue by kmaehashi** [cupy/cupy#5833]
It is a common use-case to define [PyTorch autograd functions](https://pytorch.org/docs/stable/notes/extending.html) with CuPy's RawKernel. It would be nice to cover that in the interoperability docs.

Schenario: define `torch.autograd.Function`, implement `forward` and `backward` using `cupy.RawKernel` (zero-copy using `cupy.from_dlpack(toDlpack(tensor.detach()))`), and use it from `torch.nn.Module`.

**Comment by emcastillo** [comment]
@cako #5973 shows an example of how to do this!

**Comment by cako** [comment]
Super! 💯
